src/main/java/Board.java

```diff
@@ -9,9 +12,74 @@ public Grid getGrid() {
         return grid;
     }
 
-    public void addMarkToGrid(int position, String mark) {
+    public void takeSquare(int position, String playerMark) {
         int index = position - 1;
-        grid.setSquare(index, mark);
+        grid.setSquare(index, playerMark);
+    }
+
+    public boolean moveValid
+            (int position, String player1Mark, String player2Mark) {
+
+        return positionValid(position)
+                && squareAvailable(position, player1Mark, player2Mark);
+    }
+
+    public boolean winningLine(String playerMark) {
+        var winner = false;
+        var count = 0;
+        var size = 0;
+
+        for (int[] combination : grid.getWinningCombinations()) {
+            size = combination.length;
+
+            for (int index : combination) {
```

Comment:
Do you think that refactoring lines 35-39 would have sense? It's understandable what they are doing, but it could potentially read better.

Reply:
@Kotauror it's possible. Do you have any suggestions on how it could be refactored?